java/datastructures/lib/src/test/java/datastructures/graph/GraphTest.java

```diff
@@ -81,4 +82,39 @@ public void testSingleNodeAndEdgeGraph() {
         assertEquals(singleVertex, edges.get(0).destination);
         assertEquals(0, edges.get(0).weight);
     }
+
+    @Test
+    public void testBreadthFirstEmpty() {
+        Graph<Integer> graph = new Graph<>(1);
+        List<Vertex<Integer>> result = graph.breadthFirst();
+        assertEquals(0, result.size());
+    }
+
+    @Test
+    public void testBreadthFirstSingle() {
+        Graph<String> graph = new Graph<>(1);
+        graph.addVertex("A");
+        List<Vertex<String>> result = graph.breadthFirst();
+        assertEquals(1, result.size());
+        assertEquals("A", result.get(0).value);
+    }
+
+    @Test
+    public void testBreadthFirstMultiple() {
+        Graph<Character> graph = new Graph<>(1  );
+        graph.addEdge(new Vertex<>('A'), new Vertex<>('B'));

```

Comment:
These vertices are getting created, but not added to the graph, so the test throws an error. The clunky way to do this is to manually create each vertex, add it to the graph, and then pass the vertices into your `addEdge` method. 

I bet you could also do something cool with `addEdge`, where if you pass in vertices that don't exist, it will call `addVertex` to create those as well.